Doc/library/pathlib.rst

```diff
@@ -1280,9 +1280,9 @@ call fails (for example because the path doesn't exist).
    If *exist_ok* is false (the default), :exc:`FileExistsError` is
    raised if the target directory already exists.
 
-   If *exist_ok* is true, :exc:`FileExistsError` exceptions will be
-   ignored (same behavior as the POSIX ``mkdir -p`` command), but only if the
-   last path component is not an existing non-directory file.
+   If *exist_ok* is true, :exc:`FileExistsError` is raised only if the given
+   path already exists in the file system and is not a directory (same
+   behavior as the POSIX ``mkdir -p`` command).
```

Comment:
Perhaps semantic, but I think the below is slightly more useful: as the reader, it is more important to note that when this flag is used, errors only are raised in some special case, as this flag exists precisely to suppress errors, eg in the common case of ensuring a given directory exists.

```suggestion
   If *exist_ok* is true, :exc:`FileExistsError` will not be raised unless the given
   path already exists in the file system and is not a directory (same
   behavior as the POSIX ``mkdir -p`` command).
```